VPC/main.tf

```diff
@@ -0,0 +1,135 @@
+resource "aws_vpc" "main" {
+  cidr_block           = var.vpc_cidr
+  instance_tenancy     = var.tenancy
+  enable_dns_hostnames = true
+  tags = {
+    "Name" = var.vpc_name  
+    }
+}
+
+resource "aws_internet_gateway" "IGW" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    "Name" = "${aws_vpc.main.tags.Name}-${var.Igw_name}"
+    
+      }
+
+}
+
+
+// Public Subnet Creation
+
+
+
+resource "aws_subnet" "public_subnets" {
+  for_each = var.public_subnet_config
+
+  vpc_id     = aws_vpc.main.id
+  cidr_block = each.value.cidr_block
+  availability_zone = each.value.availability_zone
+
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "${aws_vpc.main.tags.Name}-${each.key}"
+  }
+}
+
+
+resource "aws_route_table" "public_route_table" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "${aws_vpc.main.tags.Name}-Public_route_table"
```

Comment:
Why are you taking name from resource block?

Reply:
Resolved, I define the variable for the same